handler.go

```diff
@@ -98,9 +119,36 @@ func (h *handler) consume() error {
 	if err != nil {
 		return err
 	}
-	for message := range messages {
-		if _, err := h.websocket.Write(message.Body); err != nil {
-			return err
+	h.messages = msgChan
+	return nil
+}
+
+func (h *handler) setUpTicker() {
+	h.ticker = time.NewTicker(pingPeriod)
+	h.websocket.SetReadDeadline(time.Now().Add(pongWait))
+	pongHandler := func(string) error {
+		h.websocket.SetReadDeadline(time.Now().Add(pongWait));
+		return nil
+	}
+	h.websocket.SetPongHandler(pongHandler)
+}
+
+func (h *handler) consume() error {
+	defer func() {
+		h.channel.Close()
+		h.websocket.Close()
+		h.ticker.Stop()
+	}()
+	for {
+		select {
+		case message := <-h.messages:
+			if err := h.websocket.WriteMessage(websocket.TextMessage, message.Body); err != nil {
```

Comment:
TBH i have no idea. But returning `nil` instead of `err` caused panic in `http.serve` on client disconnect